File: Python/Endpoint Examples/JSON Payload/zip.py
import requests
import json

with open('/path/to/first_file', 'rb') as f: # Note that the full file name plus extension needs to be reflected in 'Content-Filename below'
    upload_first_file_data = f.read()

print("Uploading first file...")
upload_first_file_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_first_file_data,
                    headers={'Content-Type': 'application/octet-stream', 'Content-Filename': 'first_file', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("First upload response status code: " + str(upload_first_file_response.status_code))


with open('/path/to/second_file', 'rb') as f: # Note that the full file name plus extension needs to be reflected in 'Content-Filename below'
    upload_second_file_data = f.read()

print("Uploading second file...")
upload_second_file_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_second_file_data,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'second_file', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Second upload response status code: " + str(upload_second_file_response.status_code))

if upload_first_file_response.ok and upload_second_file_response.ok:
    upload_first_file_response_json = upload_first_file_response.json()
    print(json.dumps(upload_first_file_response_json, indent = 2))

    upload_second_file_response_json = upload_second_file_response.json()
    print(json.dumps(upload_second_file_response_json, indent = 2))


    uploaded_first_file_id = upload_first_file_response_json['files'][0]['id']
    uploaded_second_file_id = upload_second_file_response_json['files'][0]['id']
    zip_data = { "id" : [uploaded_first_file_id, uploaded_second_file_id] }
    print(json.dumps(zip_data, indent = 2))


    print("Processing file...")
    zip_response = requests.post(url='https://api.pdfrest.com/zip',
                        data=json.dumps(zip_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(zip_response.status_code))
    if zip_response.ok:
        zip_response_json = zip_response.json()
        print(json.dumps(zip_response_json, indent = 2))

    else:
        print(zip_response.text)
else:
    print(upload_first_file_response.text)
    print(upload_second_file_response.text)

File: JavaScript/Endpoint Examples/JSON Payload/zip.js
var axios = require("axios");
var FormData = require("form-data");
var fs = require("fs");

// By default, we use the US-based API service. This is the primary endpoint for global use.
var apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//var apiUrl = "https://eu-api.pdfrest.com";

var first_upload_data = fs.createReadStream("/path/to/first_file");

var first_upload_config = {
  method: "post",
  maxBodyLength: Infinity,
  url: apiUrl + "/upload",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
    "Content-Filename": "first_filename.pdf",
    "Content-Type": "application/octet-stream",
  },
  data: first_upload_data, // set the data to be sent with the request
};

// send request and handle response or error
axios(first_upload_config)
  .then(function (first_upload_response) {
    var first_uploaded_id = first_upload_response.data.files[0].id;

    var second_upload_data = fs.createReadStream("/path/to/second_file");

    var second_upload_config = {
      method: "post",
      maxBodyLength: Infinity,
      url: apiUrl + "/upload",
      headers: {
        "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
        "Content-Filename": "second_filename.pdf",
        "Content-Type": "application/octet-stream",
      },
      data: second_upload_data, // set the data to be sent with the request
    };

    axios(second_upload_config)
      .then(function (second_upload_response) {
        console.log(JSON.stringify(second_upload_response.data));
        var second_uploaded_id = second_upload_response.data.files[0].id;

        var zip_config = {
          method: "post",
          maxBodyLength: Infinity,
          url: apiUrl + "/zip",
          headers: {
            "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
            "Content-Type": "application/json",
          },
          data: {
            id: [first_uploaded_id, second_uploaded_id],
          }, // set the data to be sent with the request
        };

        // send request and handle response or error
        axios(zip_config)
          .then(function (zip_response) {
            console.log(JSON.stringify(zip_response.data));
          })
          .catch(function (error) {
            console.log(error);
          });
      })
      .catch(function (error) {
        console.log(error);
      });
  })
  .catch(function (error) {
    console.log(error);
  });
